src/npc.c

```diff
@@ -1331,11 +1337,42 @@ INCLUDE_ASM(void, "npc", func_8003DA38, Npc* npc, s32 arg1);
 
 INCLUDE_ASM(s32, "npc", func_8003DC38);
 
-INCLUDE_ASM(s32, "npc", func_8003DFA0);
+void func_8003DFA0(Npc* npc) {
+    f32 temp_f20;
+    f32 subroutine_argA;
+    f32 subroutine_argB;
+    u32 phi_v0;
+
+    phi_v0 = (s32) (s16) (D_80077C30++) < 4;
+    if (phi_v0 == 0) {
+        D_80077C30 = 0U;
+        temp_f20 = (clamp_angle(-npc->yaw) * TAU) / 360.0f;
+        subroutine_argA = sin_rad(temp_f20);
+        subroutine_argB = cos_rad(temp_f20);
+        playFX_0C(npc->pos.x + (npc->collisionRadius * subroutine_argA * 0.2f), npc->pos.y + 1.5f, npc->pos.z + (npc->collisionRadius * subroutine_argB * 0.2f), -npc->yaw, (f32) D_80077C34);
+        phi_v0 = D_80077C34 == 0;
+        D_80077C34 = phi_v0;
```

Comment:
`phi_v0` doesn't need to be a temp; `D_80077C34 = D_80077C34 == 0` matches by itself.